src/main/java/ch/njol/skript/expressions/ExprDistance.java

```diff
@@ -43,8 +43,12 @@ public boolean init(Expression<?>[] vars, int matchedPattern, Kleenean isDelayed
 	protected Number[] get(Event event) {
 		Location l1 = loc1.getSingle(event);
 		Location l2 = loc2.getSingle(event);
-		if (l1 == null || l2 == null || l1.getWorld() != l2.getWorld())
+		if (l1 == null || l2 == null)
 			return new Number[0];
+		if (l1.getWorld() != l2.getWorld()) {
+			error("Cannot calculate the distance between a location from " + l1.getWorld().getName() + " and " + l2.getWorld().getName() + "!");
```

Comment:
```suggestion
			error("Cannot calculate the distance between locations from two different worlds! (" + Classes.toString(l1.getWorld()) + " and " + Classes.toString(l2.getWorld()) + ")");
```
Prefer Classes.toString, since it uses the standard toString for the given skript type (in case we want to change that, we only have to change it in the type's parser, not in random expressions like this) and it properly handles the lang-file based `<none>` instead of printing `null`.

Sorry about the previous suggestion being a trap, lol, this one is proper code.